### DISCUSSION TITLE
Group iOS push notifications

### DISCUSSION THREAD

**Discussion by guevaradan** [magicbell/community#67] [Questions] [answered]
Hi there! 
Is there a way to group iOS push notifications by topic? 
We would like to group the notifications in the same way WhatsApp does for “groups” 

See below a video of what I mean:
https://user-images.githubusercontent.com/124577009/216956687-65dcaa13-94f4-4deb-b224-b8f880738448.mp4

Thanks a lot,
Dan

**Comment by niyabits** [comment, marked as answer]
Hey @guevaradan you will need to use an [iOS specific Provider Override](https://www.magicbell.com/docs/mobile-push/ios#provider-overrides) for grouping the notifications.

You will need to pass in a `thread-id` parameter which is same across the notifications you want to group together, so something like passing the topic name there should work.

Here's what your payload might look like:

```json
{
  "notification": {
    "title": "We are processing your order",
    "content": "<p>Thank you for your order. We will notify you when these items are ready.</p>",
    "recipients": [
      {
        "email": "dan@example.com"
      }
    ],
    "overrides": {
      "providers": {
        "ios": {
          "thread-id": "topic-id-123"
        }
      }
    }
  }
}
```

Apple's Documentation -https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html

**Reply by guevaradan** [reply]
Thanks a lot! I will let Victor take a look at this

**Reply by victorcerqueira** [reply]
Hey @hoppercomplex, we've just tried this out and it seems to work fine, thank you.

Just one thing, it needs to be `thread-id` and not `thread_id`  on the request payload.

**Reply by niyabits** [reply]
@victorcerqueira My bad, edited that in my answer.
